config/400-config-istio.yaml

```diff
@@ -41,25 +41,76 @@ data:
     # this example block and unindented to be in the data block
     # to actually change the configuration.
 
-    # A gateway and Istio service to serve external traffic.
-    # The configuration format should be
-    # `gateway.{{gateway_namespace}}.{{gateway_name}}: "{{ingress_name}}.{{ingress_namespace}}.svc.cluster.local"`.
-    # The {{gateway_namespace}} is optional; when it is omitted, the system will search for
-    # the gateway in the serving system namespace `knative-serving`
+
+    # external-gateways defines a gateway and Istio service to serve external traffic.
+    # It is the new and preferred way to define the configuration.
+    # The format is as follow:
+    # ```
+    # external-gateways: |
+    #   - name: {{gateway_name}}
+    #     namespace: {{gateway_namespace}}
+    #     service: {{ingress_name}}.{{ingress_namespace}}.svc.cluster.local
+    # ```
+    # name, namespace & service are mandatory and can't be empty.
+    external-gateways: |
+      - name: knative-ingress-gateway
+        namespace: knative-serving
+        service: istio-ingressgateway.istio-system.svc.cluster.local
+    #
+    #
+    # The old format has the following pattern:
+    # ```
+    # gateway.{{gateway_namespace}}.{{gateway_name}}: "{{ingress_name}}.{{ingress_namespace}}.svc.cluster.local"
+    # ```
+    # Please use the new configuration format `external-gateways` for future compatibility.
+    # This configuration format is ignored if either `external-gateways` or `local-gateways` is defined.
     gateway.knative-serving.knative-ingress-gateway: "istio-ingressgateway.istio-system.svc.cluster.local"
+    #
+    #
+    # If no values are specified for the external traffic, it will default to:
```

Comment:
Looks very good now. I think this block is not necessary. We already have more or less the "default" at line 55. Maybe we can use {{cluster_domain}} at line 52 to make this part also clear.

Reply:
Or, if we want to keep the block, it should be above the old format. It seems a bit odd to go back to the new key after the old key.

Reply:
Agree, it was done that way because it's the default value if none of the new and old format is defined.
I think I will just remove it, it's confusing wherever it is.

Reply:
Done